test/unittest/transport/SharedMemTests.cpp

```diff
@@ -1535,8 +1555,16 @@ TEST_F(SHMTransportTests, buffer_recover)
             // The segment should never overflow
             auto buf = segment->alloc_buffer(1, std::chrono::steady_clock::time_point());
 
-            ASSERT_EQ(true, pub_sub1_write->try_push(buf));
-            ASSERT_EQ(true, pub_sub2_write->try_push(buf));
+            {
+                bool is_port_ok = false;
```

Comment:
Maybe this variable can be declared out of this scope so it can be used in several checks instead of re-declaring it again. Also, consider if instead of `ASSERT` we can use `EXPECT` so the test keeps running even if some non-essential check fails.

Reply:
I'll keep the asserts. A failure trying to push to the Buffer means that there is something wrong with it and the test has to stop.